### PR TITLE
[fix] Adding SIP-15 support for the query context

### PR DESCRIPTION
**Pull request by john-bodley**
### CATEGORY

Choose one

- [x] Bug Fix
- [ ] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

This PR ensures that the query context sets the query time range endpoints if undefined and SIP-15 is enabled.

### TEST PLAN

Added unit test.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS

to: @etr2460 @villebro

### CODE REVIEW

File: tests/core_tests.py [hunk 1]
```diff
@@ -973,7 +984,12 @@ def test_results_default_deserialization(self):
             "sql": "SELECT * FROM birth_names LIMIT 100",
             "status": utils.QueryStatus.PENDING,
         }
-        serialized_data, selected_columns, all_columns, expanded_columns = sql_lab._serialize_and_expand_data(
+        (
```
Comment (john-bodley) [inline tests/core_tests.py]: This is `black` formatting.

File: tests/core_tests.py [hunk 2]
```diff
@@ -1016,7 +1032,12 @@ def test_results_msgpack_deserialization(self):
...
```
Comment (john-bodley) [inline tests/core_tests.py]: This is `black` formatting.

**Review by villebro — APPROVED** [review]
LGTM as quick fix, however I wonder if we should try to establish conventions for what should go into `extras` and what warrants a dedicated attribute, e.g. `relative_start`, `relative_end` etc, which IMO seem very much like extras. FYI @rusackas re: `viz.py` deprecation

**Comment by john-bodley** [comment]
@villbro I agree. I’m not sure what the original intent was for `extras` and whether it was merely a convenient solution to circumvent the rigidity of the function signature. This is especially true for something like the query context which would also require changes to the `superset-ui` package.

**Comment by villebro** [comment]
Looking at the current signature, I feel comfortable with most of the other parameters, but the `relative_start` and `relative_end` ones definitely belong in `extras`. You might have more context here, but I'm fairly we can move them over to `extras` without breaking anything. If not feel free to merge this as is, I can also address this later, as I'll probably be touching this code in the near future.

**Pull request merged**